salt/config/__init__.py

```diff
@@ -840,6 +840,13 @@ def _gather_buffer_space():
     # Always generate minion id in lowercase.
     'minion_id_lowercase': bool,
 
+    # The behavior when generating a minion ID. Can specify 'str' or 'func'.
+    # If 'str' is specified (the default), the 'id' option should be set to
+    # the minion ID or left unspecified for default minion ID generation
+    # behavior. If 'func' is specified, the 'id' option should be set to an
+    # exec module function to run to determine the minion ID.
+    'minion_id_type': str,
```

Comment:
The notion that somebody should have to write a function just to use a file to read a minion id strikes me as unnecessary indirection. The way that @skizunov has documented it makes it quite clear what should be done in order to achieve the desired functionality and leaves the door open for additional options.